internal/ignition/ignition.go

```diff
@@ -1435,6 +1440,24 @@ func SetHostnameForNodeIgnition(ignition []byte, host *models.Host) ([]byte, err
 	return configBytes, nil
 }
 
+func (ib *ignitionBuilder) shouldAppendOKDFiles(ctx context.Context, infraEnv *common.InfraEnv, cfg IgnitionConfig) (string, bool) {
+	// Use OKD override if OKD_RPMS_IMAGE explicitly set in config
+	if cfg.OKDRPMsImage != "" {
+		return cfg.OKDRPMsImage, true
+	}
+	// Check if selected payload contains `okd-rpms` image
+	releaseImage, err := ib.versionHandler.GetReleaseImage(infraEnv.OpenshiftVersion, infraEnv.CPUArchitecture)
```

Comment:
It's not really correct to get a release image from an infraenv version.
This version attribute is poorly named, but it is the coreos version, not an openshift version.
It's likely that this will, at some point, not match a version in the release image.

Is there a better way to detect what you need? Or is it something we can only do if we're not using late-binding?

See my comment in a very similar bug [here](https://issues.redhat.com/browse/MGMT-12482?focusedCommentId=21220244&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-21220244) for details. Essentially there's no reason for this to match any configured release image. Additionally it's valid for a user to discover hosts before adding any release image at all.

Reply:
Found this while trying to get https://github.com/openshift/assisted-service/pull/4645 ready for review FWIW

Reply:
Discussed in Slack. Its fine to leave as is, potentially late binding usecase would be broken for OKD - but this is acceptable for now